app/dmc/src/main.c

```diff
@@ -403,7 +403,11 @@ int main(void)
 				 */
 				bh_chip_cancel_bus_transfer_clear(chip);
 
-				jtag_bootrom_reset_asic(chip);
+				if (jtag_bootrom_reset_asic(chip) < 0) {
+					chip->data.performing_reset = false;
+					LOG_ERR("Failed to reset bh_chip %p", chip);
+					break;
```

Comment:
Do we want to break out here, or just return from `main` (effectively to an idle loop)? We should also enable CONFIG_LOG if we want to be able to timeout here. Otherwise, it is going to be challenging to figure out where a BMFW failed- currently you can just attach with a debugger and see that the FW is hung here.

Reply:
@danieldegrasse  I originally considered returning from main, but then our firmware would effectively do nothing in response to stimuli.

At least by using a break in the PERST handler, we can go on to run the PGOOD handler, etc. I.e. I think it's better for the firmware to be able to respond _somehow_ than to be in an infinite loop or idle and unresponsive.

Reply:
Sure, that makes sense. Break is ok with me, so long as we turn on CONFIG_LOG.

Reply:
Yikes - enabling logging adds 6k to the binary size. That seems excessive.
```
53248 is < the binary size 59052
```